src/main/java/hu/oe/nik/szfmv/automatedcar/visualization/CourseDisplay.java

```diff
@@ -129,29 +128,20 @@ private void drawObjects(Graphics2D g2d, WorldManager world) {
         DebugViewer viewer = new DebugViewer(g2d);
         AutomatedCar car = world.getAutomatedCar();
         for (IObject object : world.getAllObjectsInRectangle(
-                new hu.oe.nik.szfmv.automatedcar.model.Position(0,0),
-                new hu.oe.nik.szfmv.automatedcar.model.Position(this.world.getWidth(),this.world.getHeight()))) {
-            Point2D refPoint;
-            try {
-                refPoint = new Point(object.getReferenceX(), object.getReferenceY());
-            } catch (NullPointerException e){
-                refPoint = new Point(0,0);
-            }
-            AffineTransform t = new AffineTransform();
+                new Position(0,0),
+                new Position(this.world.getWidth(),this.world.getHeight()))) {
+            AffineTransform t = object.getTransform(offsets[0], offsets[1]);
```

Comment:
Object.getTransform offset nélkül majd utána sorban beállítod az offsetet

Reply:
És utána draw

Reply:
display-be miért nyúltok bele? :D elvileg az az a4 domain-je

Reply:
Ez a tegnapi branched javítása, azt lett egy kicsit refaktorálva. A változtatások miatt tagaltem be nálatok egy csomó embert.

Reply:
És ha megnézed a commitokat akkor láthatod, hogy csak a 3 órával ezelőtti commitban lett a courseDisplay módosítva az AffineTransform átemelésével a worldobjectbe.

Reply:
igen, azt látom és köszi az extra infókat a világ modellebe!
de ha refaktorálni kell a kódunkat, akkor írjatok nekünk és megcsináljuk :D
tudom, hogy így lassabb, de ilyen a szervezett fejlesztés sajna :)

Reply:
Persze értem mire célzol, a DebugView a futás sikeressége miatt lett módosítva, a CourseDisplay módosításáról @deseatnikov -al vitatkozz, már megint egy PR-ben akarja megcsinálni a teljes programot. :)